src/Java.Interop/Java.Interop/JniRuntime.JniTypeManager.cs

```diff
@@ -119,6 +151,37 @@ IEnumerable<JniTypeSignature> CreateGetTypeSignaturesEnumerator (Type type)
 				}
 			}
 
+			static Type GetUnderlyingType (Type type, out int rank)
+			{
+				rank = 0;
+				var originalType = type;
+				while (type.IsArray) {
+					if (type.IsArray && type.GetArrayRank () > 1)
+						throw new ArgumentException ("Multidimensional array '" + originalType.FullName + "' is not supported.", nameof (type));
+					rank++;
+					type = type.GetElementType ();
+				}
+
+				if (type.IsEnum)
+					type = Enum.GetUnderlyingType (type);
+
+				return type;
+			}
+
+
+			// `type` will NOT be an array type.
```

Comment:
We should assert this:

```csharp
Debug.Assert (!type.IsArray, $"Array type `{type}` is not supported!");
```

This won't be meaningful most of the time -- because most of the time we'll be using a Release config build, which won't contain the `Debug.Assert()` -- but it means the comment itself will be (moderately more) *actionable*.

Reply:
Wait, we throw an exception.  lol.